src/components/Files/FilesTab.tsx

```diff
@@ -0,0 +1,691 @@
+import { useQuery } from "@tanstack/react-query";
+import dayjs from "dayjs";
+import { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import CareIcon, { IconName } from "@/CAREUI/icons/CareIcon";
+
+import { Badge } from "@/components/ui/badge";
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { Progress } from "@/components/ui/progress";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { Tabs, TabsContent } from "@/components/ui/tabs";
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from "@/components/ui/tooltip";
+
+import AudioPlayer from "@/components/Common/AudioPlayer";
+import Loading from "@/components/Common/Loading";
+import TextFormField from "@/components/Form/FormFields/TextFormField";
+import { FileUploadModel } from "@/components/Patient/models";
+
+import useFileManager from "@/hooks/useFileManager";
+import useFileUpload, { FileUploadReturn } from "@/hooks/useFileUpload";
+import useFilters from "@/hooks/useFilters";
+
+import { FILE_EXTENSIONS } from "@/common/constants";
+
+import routes from "@/Utils/request/api";
+import query from "@/Utils/request/query";
+import { classNames } from "@/Utils/utils";
+import { usePermissions } from "@/context/PermissionContext";
+
+export interface FilesTabProps {
+  type: "encounter" | "patient";
+  facilityId: string;
+  encounterId?: string;
+  patientId?: string;
+}
+
+export const FilesTab = (props: FilesTabProps) => {
+  const { encounterId, patientId, type } = props;
+  const { qParams, updateQuery, Pagination, resultsPerPage } = useFilters({
+    limit: 14,
+  });
+  const { t } = useTranslation();
+  const [openUploadDialog, setOpenUploadDialog] = useState(false);
+  const [selectedAudioFile, setSelectedAudioFile] =
+    useState<FileUploadModel | null>(null);
+  const [openAudioPlayerDialog, setOpenAudioPlayerDialog] = useState(false);
+  const { hasPermission } = usePermissions();
+
+  const associatingId =
+    {
+      patient: patientId,
+      encounter: encounterId,
+    }[type] || "";
+
+  const fileCategories = [
+    { value: "all", label: "All" },
+    { value: "imaging", label: "Imaging" },
+    { value: "lab_reports", label: "Lab Reports" },
+    { value: "documents", label: "Documents" },
+    { value: "audio", label: "Audio" },
+  ];
+
+  const handleTabChange = (value: string) => {
+    updateQuery({ file_category: value === "all" ? undefined : value });
+  };
+
+  const {
+    data: files,
+    isLoading: filesLoading,
+    refetch,
+  } = useQuery({
+    queryKey: [type + "-files", associatingId, qParams],
```

Comment:
```suggestion
    queryKey: [files, type, associatingId, qParams],
```

Reply:
🤔  Ah, good catch..had done that to ensure patient and encounter files queries don't mix...but the associating Id for both should be different so it won't be an issue anyways 👍